python/amici/petab_objective.py

```diff
@@ -124,6 +124,10 @@ def simulate_petab(
         parameter_mapping=parameter_mapping,
         scaled_parameters=scaled_parameters)
 
+    # only required if we have preequilibration AND species or
+    #  compartments in condition table, but shouldn't hurt to enable anyways.
+    amici_model.setReinitializeFixedParameterInitialStates(True)
```

Comment:
doesn't hurt to always enable? are there any performance drawbacks?

Reply:
Wouldn't expect anything noticeable. Didn't time. Can be optimized later.

Reply:
Its not about performance, more about validity. You may not always want to reinitialize your model after preequilibration.

Reply:
Hmm... 
What's was again the motivation for having to set that manually? Couldn't this be handled internally, depending on whether `fixedParametersPreequilibration` are set? In which case would I want to enable that if I don't change fixedParameters after preequilibration? In which cases  would I want to disable that if do change them?

Reply:
What about adding that to `ExpData`? Otherwise it's not possible to use runAmiciSimulation**s** for a set of conditions where some need reinitialization and and some don't.

Reply:
Might be that this was added when not all model were supporting reinitialization. Adding to ExpData makes sense. FixedParametersPreequilibration may also set dynamic fixedParameters, so I dont think we always want to enforce if they are set.

Reply:
Okay, will merge this PR as is, and address this separately together with changes in ExpData.